### PR TITLE
Arm

### PR DESCRIPTION
**Pull request by SolidityContract**
Added Arm Subsystem

### CODE REVIEW

File: src/main/java/org/troyargonauts/Constants.java [hunk 1]
```diff
@@ -7,4 +7,19 @@ public interface Manipulator {
         public static final int kReverseChannel = 1;
     }
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/Constants.java:8]: REMOVE manipulator interface from this branch

File: src/main/java/org/troyargonauts/subsystems/Arm.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        wristMotor.setIdleMode(CANSparkMax.IdleMode.kBrake);
+
+        elbowEncoder = elbowMotor.getAbsoluteEncoder(SparkMaxAbsoluteEncoder.Type.kDutyCycle);
+        wristEncoder = wristMotor.getAbsoluteEncoder(SparkMaxAbsoluteEncoder.Type.kDutyCycle);
+
+        pid = new PIDController(Constants.Arm.kP, Constants.Arm.kI, Constants.Arm.kD, Constants.Arm.PERIOD);
+    }
+
+    /**
+     * Enums are the states of the intake rollers.
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/Arm.java:43]: What are the states? List them in javadoc

File: src/main/java/org/troyargonauts/subsystems/Arm.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+     */
+    public enum intakeState {
+        FORWARD, OFF, BACKWARD
+    }
+
+    /**
+     * Intake roller state is set here.
+     *
+     * @param state desired state of intake rollers (FORWARD, OFF, BACKWARD) to a motor speed
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/Arm.java:52]: Remove list of states from parameter

File: src/main/java/org/troyargonauts/subsystems/Arm.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+     * Activates intake motors to BACKWARD state
+     */
+    public void reverseIntake() {
+        intakeMotor.set(Constants.Arm.REVERSE_INTAKE_SPEED);
+    }
+
+    /**
+     * Sets Elbow to set speed given it is within encoder limits
+     *
+     * @param speed sets elbow motor to desired speed given that it is within the encoder limits
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/Arm.java:89]: "desired motor speed"

File: src/main/java/org/troyargonauts/subsystems/Arm.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        }
+        else {
+            elbowMotor.set(speed);
+        }
+    }
+
+    /**
+     * Sets Wrist to set speed given it is within encoder limits
+     *
+     * @param speed sets wrist motor to desired speed given that it is within the encoder limits
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/Arm.java:103]: "desired motor speed"

File: src/main/java/org/troyargonauts/subsystems/Arm.java [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+     * Returns wrist position
+     *
+     * @return returns the position of wrist
+     */
+    public double getWristPosition() {
+        return wristEncoder.getPosition();
+    }
+
+    /**
+     * Using a PID command, elbow will rotate to a setpoint using the PID Controller. Will mainly be used in autonomous.
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/Arm.java:133]: Remove "will mainly be used in autonomous"

File: src/main/java/org/troyargonauts/subsystems/Arm.java [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+     * @param setpoint the point desired to be reached by the elbow
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/Arm.java:134]: "desired setpoint"

File: src/main/java/org/troyargonauts/subsystems/Arm.java [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+                pid,
+                this::getElbowPosition,
+                setpoint,
+                this::setElbowPower,
+                Robot.getArm()
+        );
+    }
+
+    /**
+     * Using a PID command, wrist will rotate to a setpoint using the PID Controller. Will mainly be used in autonomous.
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/Arm.java:148]: remove "will mainly be used in autonomous"

File: src/main/java/org/troyargonauts/subsystems/Arm.java [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+     * @param setpoint the point desired to be reached by the wrist
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/Arm.java:149]: "desired setpoint"

**Review by kelmalon — CHANGES_REQUESTED** [review]
Please reach out with any questions

**Review by kelmalon — CHANGES_REQUESTED** [review]
Please add variables to store absolute encoder values, and update them in periodic. Please also put the values to SmartDashboard in periodic.

**Review by kelmalon — APPROVED** [review]
Looks good!

File: src/main/java/org/troyargonauts/subsystems/Arm.java [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+        wristEncoderValue = wristEncoder.getPosition();
+
+        SmartDashboard.putNumber("Elbow Encoder", elbowEncoderValue);
+        SmartDashboard.putNumber("Wrist Encoder", wristEncoderValue);
+    }
+
+    /**
+     * Enums are the states of the intake rollers (FORWARD, OFF, BACKWARD).
+     */
+    public enum intakeState {
```
Comment (DarkMG1) [inline src/main/java/org/troyargonauts/subsystems/Arm.java:56]: Please make this "IntakeState"

**Review by DarkMG1 — CHANGES_REQUESTED** [review]

**Review by DarkMG1 — APPROVED** [review]

**Review by akashsarada — APPROVED** [review]
Looks good!

**Pull request merged**